File: back/src/main/java/com/dai5/back/model/order/Payment.java
package com.dai5.back.model.order;

import com.dai5.back.model.user.User;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.Instant;

@Getter
@Setter
@Entity
@Table(name = "Payments", schema = "fodmapStore")
public class Payment {
    @Id
    @Column(name = "id", nullable = false)
    private Integer id;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "order_id", nullable = false)
    private Order order;

    @Column(name = "amount", precision = 10, scale = 2)
    private BigDecimal amount;

    @Column(name = "paid_at", nullable = false)
    private Instant paidAt;

    @Lob
    @Column(name = "payment_method")
    private String paymentMethod;

    @Lob
    @Column(name = "status")
    private String status;

}

File: front/src/model/order/payment/payment.ts
import { User } from "src/model/user/user";
import { Order } from "../order";
import { PaymentStatus } from "./paymentStatus.enum";
import { PaymentMethod } from "./paymentMethod.enum";

export interface Payment {
    id: number;
    user: User;
    order: Order;
    amount: number;
    paidAt: Date;
    paymentStatus: PaymentStatus;
    paymentMethod: PaymentMethod;
}